core/design/src/main/kotlin/edu/stanford/spezi/core/design/component/ImageResource.kt

```diff
@@ -0,0 +1,19 @@
+package edu.stanford.spezi.core.design.component
+
+import androidx.annotation.DrawableRes
+import androidx.compose.ui.graphics.vector.ImageVector
+import edu.stanford.spezi.core.utils.UUID
+
+/**
+ * A sealed class to represent an image resource. It can be either a vector image or a drawable.
+ * This is useful to abstract the image resource type and use it in a composable function. The identifier can be used for tests.
+ * @see ImageResource.Vector
+ * @see ImageResource.Drawable
+ * @see ImageResourceIcon
+ */
+sealed class ImageResource(open val identifier: String = UUID().toString()) {
```

Comment:
This property is declared open, but none of the types is overriding it, and I also see no need to override. Feel free to move it out of constructor as a property of the sealed class itself:

```kotlin
sealed class ImageResource { 
       val identifier: String = UUID().toString()
       // other types
}
```

Even though we have one single usage for loading an image via a url (ExpandableVideoSection.kt) where we use `SubcomposeAsyncImage`, I would add a third case in this sealed class, and ideally replace the usage there as well via the model

```kotlin
data class Remote(val url: String) : ImageResource()
```

Reply:
An additional comment, please also annotate the ImageResource with `@Immutable` - an additional hint to compose to avoid unnecessary recompositions because we guarantee that the instance won't change at runtime.

Reply:
Yes, that makes sense. I still had it like that because I thought it might make sense to overwrite it, for example to set the url as identifier or the resIdltoString but I think it fits to just pull it out from the constructor 👍